### PR TITLE
[indexer] Comment geometry scale levels code and refactor it a bit

### PR DESCRIPTION
**Pull request by pastk**
Mostly from https://github.com/organicmaps/organicmaps/pull/2798.

### CODE REVIEW

File: indexer/drules_selector.cpp [hunk 1]
```diff
@@ -153,6 +153,8 @@ bool GetBoundingBoxArea(FeatureType & ft, double & sqM)
   if (feature::GeomType::Area != ft.GetGeomType())
     return false;
 
+  // FIXME: the best geometry is loaded here always, use the current zoom level
+  // see https://github.com/organicmaps/organicmaps/issues/2840
   sqM = mercator::AreaOnEarth(ft.GetLimitRect(scales::GetUpperScale()));
```
Comment (vng) [inline indexer/drules_selector.cpp:158]: Hm, good catch. In fact, we killed the initial different zooms idea here.
I see that zoom is already passed into BaseRule::TestFeature. Can you restore it here?

Comment (pastk) [reply indexer/drules_selector.cpp:158]: I looked at it awhile ago, so don't remember the details atm..

**Review by pastk — COMMENTED** [review]
@vng your commit looks good, I tested it fixes #2840!

**Comment by vng** [comment]
Ok, will merge it after release. Put approve :)

**Review by vng — APPROVED** [review]

**Pull request merged**